src/main/scala/nl.knaw.dans.easy.multideposit/actions/ReportDatasets.scala

```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright (C) 2016 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.multideposit.actions
+
+import better.files.{ ManagedResource, _ }
+import nl.knaw.dans.easy.multideposit.PathExplorer.OutputPathExplorer
+import nl.knaw.dans.easy.multideposit.actions.ReportDatasets._
+import nl.knaw.dans.easy.multideposit.encoding
+import nl.knaw.dans.easy.multideposit.model.Deposit
+import org.apache.commons.csv.{ CSVFormat, CSVPrinter }
+
+import scala.util.Try
+
+class ReportDatasets {
+
+  def report(deposits: Seq[Deposit])(implicit output: OutputPathExplorer): Try[Unit] = Try {
+    for(printer <- csvPrinter(output.reportFile);
```

Comment:
why `;`? Isn't this a for comprehension?

Reply:
it won't work without!
Notice that this is not a for comprehension: that's with `{...}` and a `yield`, while this is with `(...)` and without a `yield`.

Both this for loop and the for comprehension are syntactic sugar in Scala. The latter rewrites to a series of `map`, `filter` and `flatMap` calls. The former is just a nested set of `foreach` calls.

https://github.com/DANS-KNAW/easy-split-multi-deposit/blob/eb1f9e279055ffafac7959089f0b0a19fa2f917d/src/main/scala/nl.knaw.dans.easy.multideposit/actions/ReportDatasets.scala#L29-L33

In this instance, `csvPrinter(output.reportFile)` returns a `ManagedResource[CSVPrinter]`, for which the `foreach` (1) opens the resource, (2) uses the resource and (3) closes the resource. On the other hand, `deposits` is a `Seq[Deposit]`, so `foreach` iterates over all elements in the collection and performs an action for each of them.

When writing this without for-loop and with `foreach`, it would look like stated below, which probably immediately shows why it's better readable to use a for-loop.

```scala
def report(deposits: Seq[Deposit])(implicit output: OutputPathExplorer): Try[Unit] = Try {
  csvPrinter(output.reportFile).foreach(printer => deposits.foreach(deposit => printRecord(deposit, printer)))
}
```

Finally, notice that the order of the 'resources' in the for-loop matters a whole lot! If the order was reversed, you would create as many `CSVPrinter`s as there are `Deposit`s in the collection!